query.go

```diff
@@ -149,15 +149,19 @@ func getFrames(rows *sql.Rows, limit int64, converters []sqlutil.Converter, fill
 	if err != nil {
 		return nil, err
 	}
-	if count == 0 {
-		return nil, ErrorNoResults
-	}
+
+	// the handling of zero-rows differs between various "format"s.
+	zeroRows := count == 0
 
 	frame.Meta.ExecutedQueryString = query.RawSQL
 	frame.Meta.PreferredVisualization = data.VisTypeGraph
 
 	switch query.Format {
 	case FormatOptionMulti:
+		if zeroRows {
+			return nil, ErrorNoResults
```

Comment:
do we have to return an error?  or should we just return an empty frame with a warning?

Reply:
this does not cause an error "in the browser". this is only used in the go-code, and eventually it reaches here:
https://github.com/grafana/sqlds/blob/6631da79fa49982fd8275d944a8b0ffd61ef546c/datasource.go#L249-L251

so no error is returned. (the dataframe at that point is a more&less empty one, containing the executed-query-string).

personally i'd love to adjust the code to not have this "marker" error, but not sure what others think. also wanted to minimize the size of the diff. 
WDYT?